shared/src/simulation/activities/reassign-treatments.ts

```diff
@@ -0,0 +1,632 @@
+    }
+
+    const availablePersonnel = groupedPersonnel[minType]?.find(
+        (personnel) =>
+            sumOfTreatments(personnel) < maxPatients &&
+            couldCaterFor(
+                patientStatus,
+                personnel.personnel,
+                personnel.catersFor
+            ) &&
+            (mixWithHigherStatus ||
+                hasNoHigherTreatments(personnel, patientStatus))
+    );
+
+    if (availablePersonnel) {
+        return { personnel: availablePersonnel, isExclusive: false };
+    }
+
+    switch (minType) {
+        case 'gf':
+            return findAssignablePersonnel(
+                groupedPersonnel,
+                'san',
+                patientStatus,
+                maxPatients
+            );
+        case 'san':
+            return findAssignablePersonnel(
+                groupedPersonnel,
+                'rettSan',
+                patientStatus,
+                maxPatients
+            );
+        case 'rettSan':
+            return findAssignablePersonnel(
+                groupedPersonnel,
+                'notSan',
+                patientStatus,
+                maxPatients
+            );
+        case 'notSan':
+            return findAssignablePersonnel(
+                groupedPersonnel,
+                'notarzt',
+                patientStatus,
+                maxPatients
+            );
+        case 'notarzt':
+            return undefined;
+    }
+}
+
+/**
+ * Performs the actual assignment of personnel and material to patients.
+ * This function tries to reach matching in which every red patient is treated by a notSan and a rettSan exclusively, each yellow patient is treated by a rettSan exclusively and each two green patients are treated by a san.
+ * Additionally, each red and yellow patient should be treated by a notarzt (non-exclusively) and material should be assigned.
+ * @param draftState The state to operate in
+ * @param patients A list of the patients to operate on
+ * @param cateringPersonnel A list of the personnel to operate on.
+ *  Personnel may be treating some other patients already, expressed by the catersFor property.
+ * @param materials A list of the materials to operate on.
+ * @returns Whether the treatment for all patients is secured.
+ */
+function assignTreatments(
+    draftState: Mutable<ExerciseState>,
+    patients: Mutable<Patient>[],
+    cateringPersonnel: CateringPersonnel[],
+    materials: Mutable<Material>[]
+): boolean {
+    const groupedPatients = groupBy(patients, (patient) =>
+        Patient.getVisibleStatus(
+            patient,
+            draftState.configuration.pretriageEnabled,
+            draftState.configuration.bluePatientsEnabled
+        )
+    );
+
+    const groupedPersonnel = groupBy(
+        cateringPersonnel,
+        (personnel) => personnel.personnel.personnelType
+    );
+
+    let securedPatients = 0;
+
+    groupedPatients.red?.forEach((patient) => {
+        const notSanResult = findAssignablePersonnel(
```

Comment:
I dont like the Name... maybe name it notSanFound or assignableNotSan something Same for the variables below

Reply:
Did it in [0612587](https://github.com/hpi-sam/digital-fuesim-manv/pull/670/commits/06125872953c08fac27aedc0c38d5ec48832a110)